src/cubic-b-spline.cc

```diff
@@ -377,19 +377,26 @@ namespace trajectory
 #endif //! ROBOPTIM_DO_NOT_CHECK_ALLOCATION
 
     const std::size_t nbp = static_cast<std::size_t> (nbp_);
+    const size_type n = outputSize ();
+    const size_type offset = n - 1;
 
     // Cubic B-spline ---> nbp_-3 intervals (aka segments or bays)
     if (res.size () != nbp - 3)
       res.resize (nbp - 3);
 
+    if (n >= 3)
+    {
+      std::cerr << "Beware: result will very probably be false" << std::endl;
```

Comment:
A throw with `untested feature` or something may be better than printing to `cerr`.